packages/pigeon/tool/shared/generation.dart

```diff
@@ -48,60 +48,79 @@ String _javaFilenameForName(String inputName) {
 }
 
 Future<int> generateExamplePigeons() async {
-  return runPigeon(
+  int success = 0;
+  success = await runPigeon(
     input: './example/app/pigeons/messages.dart',
     basePath: './example/app',
     suppressVersion: true,
   );
+  success = await runPigeon(
+    input: './example/app/pigeons/event_channel_messages.dart',
+    basePath: './example/app',
+    suppressVersion: true,
+  );
+  return success;
 }
 
 Future<int> generateTestPigeons(
     {required String baseDir, bool includeOverflow = false}) async {
   // TODO(stuartmorgan): Make this dynamic rather than hard-coded. Or eliminate
   // it entirely; see https://github.com/flutter/flutter/issues/115169.
-  const List<String> inputs = <String>[
-    'background_platform_channels',
-    'core_tests',
-    'enum',
-    'flutter_unittests', // Only for Dart unit tests in shared_test_plugin_code
-    'message',
-    'multiple_arity',
-    'non_null_fields',
-    'null_fields',
-    'nullable_returns',
-    'primitive',
-    'proxy_api_tests',
-  ];
+
+  /// A list of all pigeons to be generated along with a list of skipped languages.
+  const Map<String, Set<GeneratorLanguage>> inputs =
+      <String, Set<GeneratorLanguage>>{
+    'background_platform_channels': <GeneratorLanguage>{},
+    'core_tests': <GeneratorLanguage>{},
+    'enum': <GeneratorLanguage>{},
+    'event_channel_tests': <GeneratorLanguage>{
+      GeneratorLanguage.cpp,
+      GeneratorLanguage.gobject,
+      GeneratorLanguage.java,
+      GeneratorLanguage.objc,
+    },
+    'flutter_unittests':
+        <GeneratorLanguage>{}, // Only for Dart unit tests in shared_test_plugin_code
+    'message': <GeneratorLanguage>{},
+    'multiple_arity': <GeneratorLanguage>{},
+    'non_null_fields': <GeneratorLanguage>{},
+    'null_fields': <GeneratorLanguage>{},
+    'nullable_returns': <GeneratorLanguage>{},
+    'primitive': <GeneratorLanguage>{},
+    'proxy_api_tests': <GeneratorLanguage>{},
+  };
```

Comment:
It would probably be easier to keep the set as-is, and then have a separate to-skip map with entries only where the skip set is non-empty, since we expect skipping to be exceptional rather than something most entries in the set will need.